listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -611,6 +640,119 @@ def create_personal_recommendation_event(user_name):
+            raise APIBadRequest(
+                f"You cannot recommend tracks to non-followers! These people don't follow you {str(non_followers)}")
+        event = db_user_timeline_event.create_personal_recommendation_event(
+            db_conn, user['id'], metadata)
+    except pydantic.ValidationError as e:
+        raise APIBadRequest(f"Invalid metadata: {str(e)}")
+    except DatabaseException:
+        raise APIInternalServerError("Something went wrong, please try again.")
+
+    event_data = event.dict()
+    event_data['created'] = int(event_data['created'].timestamp())
+    event_data['event_type'] = event_data['event_type'].value
+    return jsonify(event_data)
+
+
+@user_timeline_event_api_bp.post('/user/<user_name>/timeline-event/create/thanks')
+@crossdomain
+@ratelimit()
+def create_thanks_event(user_name):
+    '''
+    Make the user recommend a recording to their followers. The request should post
+    the following data about the recording being recommended (either one of recording_msid
+    or recording_mbid is sufficient), and also the list of followers getting recommended:
+
+    .. code-block:: json
+
+        {
+            "metadata": {
+                "recording_msid": "<The MessyBrainz ID of the recording, optional>",
+                "recording_mbid": "<The MusicBrainz ID of the recording>",
+                "users": [], // "<usernames of the persons you want to recommend to, required>"
+                "blurb_content": "<String containing personalized recommendation>"
+            }
+        }
+
+    :reqheader Authorization: Token <user token>
+    :reqheader Content-Type: *application/json*
+    :statuscode 200: Successful query, recording has been recommended!
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized, you do not have permissions to recommend
+        personal recordings on the behalf of this user
+    :statuscode 403: Forbidden, you do not have permissions to recommend
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    '''
+
+    user = validate_auth_header()
+
+    if user_name != user['musicbrainz_id']:
+        raise APIForbidden(
+            "You don't have permissions to hide events from this user's timeline.")
+
+    try:
+        data = orjson.loads(request.get_data())
+    except (ValueError, KeyError) as e:
+        raise APIBadRequest(f"Invalid JSON: {str(e)}")
+
+    if 'metadata' not in data:
+        raise APIBadRequest("JSON document must contain metadata", data)
+
+    metadata = data["metadata"]
+
+    if 'original_event_type' not in metadata or 'original_event_id' not in metadata:
+        raise APIBadRequest(
+            "Metadata must contain both original_event_type and original_event_id", metadata)
+
+    try:
+        validate = ThanksMetadata(**metadata)
+        row_id = metadata["original_event_id"]
+        event_type = metadata["original_event_type"]
```

Comment:
After the change above to use the ThanksMetadata class, you'll need to access properties like so:
```suggestion
        row_id = metadata.original_event_id
        event_type = metadata.original_event_type
```